Marlin/src/Marlin.cpp

```diff
@@ -666,6 +666,14 @@ void idle(
     bool no_stepper_sleep/*=false*/
   #endif
 ) {
+
+  #if ENABLED(SPI_ENDSTOPS)
+    if ((endstops.tmc_spi_homing.x | endstops.tmc_spi_homing.y | endstops.tmc_spi_homing.z))
+      if (ELAPSED(millis(), sg_guard_period))
+        for (uint8_t i = 4; i>0; i--) // Read SGT 4 times per idle loop
```

Comment:
Should probably abort from loop if triggered...

Reply:
So, if `live_state` is not zero after the call to `endstops.tmc_spi_homing_check`?

Reply:
Or have `tmc_spi_homing_check` return a boolean of whether or not a stallGuard endstop was triggered.
Probably can't check against live_state if the user has mixed endstop configuration and so a triggered Y would prevent checking X.